### DISCUSSION TITLE
Which dockers are needed for self hosted/performed algo/machine learning image recognition?

### DISCUSSION THREAD

**Discussion by DuckersMcQuack** [immich-app/immich#9520] [Q&A] [answered]
I want to host my images on immich on my server, and found out i can use intel hd graphics/tensor acceleration for scanning and finding images with subject i search for, and got 2 questions:

1: Can i train the model(s) myself as well if it can't detect a specific search criteria?
2: What dockers do i need to co-exist with eachother for this all to work out?
3: How much video memory is recommended to have for the trained model(s) if any to cache in?

**Comment by bo0tzz** [comment, marked as answer]
1. The short answer is no
2. If you want to use ML in Immich then you will need all the containers
3. A few gigabytes I think? @mertalev?

**Reply by mertalev** [reply]
4gb VRAM or higher should generally be fine, but it depends on the model and what else the GPU is doing.

You can get away with 3gb, but you'll be pushing the VRAM close to its limit.
